test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -191,6 +191,18 @@ static void ssl_test_setup_handshakers(tsi_test_fixture* fixture) {
       TSI_OK);
 }
 
+static void check_ca_cert_subject(ssl_tsi_test_fixture* /*ssl_fixture*/,
```

Comment:
IIUC, we are only testing in the case when verification is successful and there are no intermediate CAs in the chain. I think it would be good to test in the following cases as well:

1. When verification fails (e.g. we can use the self-signed cert in the /src/core/tsi/test_creds directory).
2. When verification succeeds but there is >=1 intermediate CA in the chain. (If there are no examples of this in /src/core/tsi/test_creds let me know and I can help you generate them.)

Reply:
Added nullptr checks in the test for cases where this should be unset

Is case 2 a special case? The implementation just gets the last cert in the verified chain, should it matter if there are intermediate CAs?

Reply:
Maybe I misunderstood, but I didn't see any tests where verification fails?

I think 2 is important because the intermediates are sent by the peer but the root is not, it's from our trust store.

Reply:
There are failed verifications in the [CRL related SSL tests](https://github.com/grpc/grpc/blob/master/test/core/tsi/crl_ssl_transport_security_test.cc)

I'll happily add more tests though, I'll discuss with you offline about setting up the chain for 2.